test/exe/pie_test.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+ENVOY_BIN=${TEST_SRCDIR}/envoy/source/exe/envoy-static
+
+if [[ `uname` == "Darwin" ]]; then
+  echo "Skipping on macOS."
+  exit 0
+fi
+
+if readelf -hW ${ENVOY_BIN} | grep "Type" | grep -o "DYN (Shared object file)"; then
+  echo "${ENVOY_BIN} is a PIE!"
+  exit 0
+fi
+
+echo "${ENVOY_BIN} is not a PIE!"
+exit 1
```

Comment:
I assume you verified this with before/after binaries..

Reply:
yes